docs/discovery-handlers/onvif.md

```diff
@@ -156,6 +361,65 @@ helm install akri akri-helm-charts/akri \
+The schema of how the Secret/ConfigMap files are organized aligned to the schema that Onvif Discovery Handler used for passing the secret data, as follow:
+1. The sample broker first looks for files under `CREDENTIAL_DIRECTORY` with file name "username_<device_uuid>" for username and "password_<device_uuid>" for password, where device_uuid is the device uuid string that replaces all '-' with underscore '_'.
+If file "username_<device_uuid>" exists, but "password_<device_uuid>" doesn't exist, the sample broker uses empty string as password. The sample broker also looks for
+files "username_default" and "password_default", if found, the username and password is used as fallback username/password if a credential with matched device uuid can
+not be found.
+2. If the sample broker can not find a matched credential from the username/password secret files directly, it looks for credentials from the device credential ref list `device_credential_ref_list`.  The sample broker try to get a file name `device_credential_ref_list` from `CREDENTIAL_CONFIGMAP_DIRECTORY`, and if the file does not 
+exists, it tries to get the same file name under `CREDENTIAL_DIRECTORY`.  The credential ref list should contains reference entries to the actual credentals. The sample
+broker look up the device uuid from the list to get the credential referece and read the actual credential files from `CREDENTIAL_DIRECTORY`.  Similar to the fallback credenial "username_default" and "password_default", a credential ref entry with key "default" indicates the fallback credental.
+3. if the sample broker can not find a matched credental from credential ref list, it looks for credentials from crdential list `device_credential_list`.  The sample broker try to get a file name `device_credential_list` from `CREDENTIAL_CONFIGMAP_DIRECTORY`, and if the file does not xxists, it tries to get the same file name under `CREDENTIAL_DIRECTORY`.  The credential list should contains actual credential entries. The sample broker look up the device uuid from the list to get the credential. Similarly, a credential entry with key "default" indicates the fallback credental.
+
+The following example shows how the credential information is organized in Secret and ConfigMap.  There are 4 credentials specified in this example, credential for device id
+"6a67158b-42b1-400b-8afe-1bec9a5d7919", "3fa1fe68-b915-4053-a3e1-ac15a21f5f91", "6a67158b-42b1-400b-8afe-1bec9a5d7909" and a fallback credential "username_default"/"password_default".
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: onvif-auth-configmap
+data:
+  device_credential_ref_list: |+ 
+    [ "credential_ref_list1" ]
+  credential_ref_list1: |+
+    {
+        "6a67158b-42b1-400b-8afe-1bec9a5d7919":
+            {
+                "username_ref" : "device2_username",
```

Comment:
should device2_username just be username?

Reply:
(same for password)

Reply:
i'm probably just missing how device2_username connects to username_6a67158b_42b1_400b_8afe_1bec9a5d7909.  maybe they don't?

Reply:
good catch, device2_username and device2_password are missing, I'll add them back

Reply:
fix in https://github.com/project-akri/akri-docs/pull/89